framework/tlvf/src/src/CmduMessage.cpp

```diff
@@ -10,149 +10,76 @@
 
 using namespace ieee1905_1;
 
-CmduMessage::CmduMessage() {}
+const uint16_t CmduMessage::kCmduHeaderLength = 8;
+const uint16_t CmduMessage::kTlvHeaderLength  = 3;
+// CMDUs are fragmanted on TLV boundary so they can't be larger than the MTU.
+// In most cases the MTU is 1500 - so use this value as the default.
+const size_t CmduMessage::kMaxCmduLength = 1500;
 
-CmduMessage::~CmduMessage() {}
+CmduMessage::CmduMessage(uint8_t *buff, size_t buff_len) : tlvs(buff, buff_len) {}
 
-std::shared_ptr<cCmduHeader> CmduMessage::getCmduHeader() const { return m_cmdu_header; }
+CmduMessage::~CmduMessage() {}
 
-std::shared_ptr<BaseClass> CmduMessage::getClass(size_t idx) const
+std::shared_ptr<cCmduHeader> CmduMessage::getCmduHeader() const
 {
-    if (m_class_vector.size() > idx) {
-        return std::shared_ptr<BaseClass>(m_class_vector.at(idx));
-    } else {
-        return nullptr;
-    }
+    return tlvs.getClass<cCmduHeader>();
```

Comment:
IMHO A TLV list is just a wrapper for TLVF classes which do not necessarily need to be TLVs. With this assumption, and the fact that an outer TLV list is only used for 1905 TLVs we can decide that the first element is always the CMDU header. I think it makes the code simpler.

Reply:
beerocks_header derives from TlvList and its classes also don't have type and length. Maybe a better name would be TlvfClassList since it's a container for TLVF classes more than it is a TLV list. Then if we really want to, we can always extend it and create a strict TLV only list.
But I think that at the moment we shouldn't restrict TlvList to contain only TLVs.

Reply:
> beerocks_header derives from TlvList and its classes also don't have type and length. 

Good point, I didn't realize that. I was too fixated on WSC :-)

> Maybe a better name would be TlvfClassList since it's a container for TLVF classes more than it is a TLV list.

That's a different idea than what I originally had, but makes a lot more sense. Indeed, if you look at it as a sequence of tlvf-generated structures, it makes a lot more sense.

So yes, a rename would be a good idea. ClassList is maybe more appropriate, like we have BaseClass.